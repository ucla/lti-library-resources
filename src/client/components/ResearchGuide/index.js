import React from 'react';
import { Button } from '@instructure/ui-buttons';
import * as constants from '../../constants';

const ResearchGuide = ({ isUserAdmin, isUserTeacher, idToken }) => (
  <div>
    {(isUserAdmin(idToken) || isUserTeacher(idToken)) && (
      <Button
        color="secondary"
        margin="small"
        href="http://ucla.libsurveys.com/rg-feedback"
        target="_blank"
        rel="noopener roreferrer"
      >
        Library resources feedback
      </Button>
    )}
  </div>
);

export default ResearchGuide;
