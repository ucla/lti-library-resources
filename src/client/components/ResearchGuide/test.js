import React from 'react';
import { shallow } from 'enzyme';
import ResearchGuide from '.';

it('Correctly displays button for teacher/admin', () => {
  const wrapper = shallow(
    <ResearchGuide isUserAdmin={() => false} isUserTeacher={() => true} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('Correctly hides button for student', () => {
  const wrapper = shallow(
    <ResearchGuide isUserAdmin={() => false} isUserTeacher={() => false} />
  );
  expect(wrapper).toMatchSnapshot();
});
