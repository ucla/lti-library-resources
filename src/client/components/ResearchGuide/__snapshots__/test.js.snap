// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly displays button for teacher/admin 1`] = `
<div>
  <Button
    as="button"
    color="secondary"
    cursor="pointer"
    display="inline-block"
    elementRef={[Function]}
    href="http://ucla.libsurveys.com/rg-feedback"
    margin="small"
    rel="noopener roreferrer"
    size="medium"
    target="_blank"
    textAlign="center"
    type="button"
    withBackground={true}
  >
    Library resources feedback
  </Button>
</div>
`;

exports[`Correctly hides button for student 1`] = `<div />`;
